armi/reactor/converters/uniformMesh.py

```diff
@@ -162,37 +163,55 @@ def makeAssemWithUniformMesh(sourceAssem, newMesh):
             sourceBlock = None
             specialXSType = None
-            for potentialBlock, _overlap in overlappingBlockInfo:
+            runLog.debug(
+                f"From axial elevation {bottom:<6.2f} cm to {topMeshPoint:<6.2f} cm"
+            )
+            for potentialBlock, heightOverlap in overlappingBlockInfo:
                 if sourceBlock is None:
                     sourceBlock = potentialBlock
+                    sourceBlockHeightOverlap = heightOverlap
                 if (
                     potentialBlock.hasFlags([Flags.FUEL, Flags.CONTROL])
                     and potentialBlock != sourceBlock
                 ):
-                    runLog.important(
-                        "There are multiple overlapping blocks.  Choosing {} for {} XS sets.".format(
-                            potentialBlock.getType(), sourceBlock.getType()
-                        )
+                    totalHeight = topMeshPoint - bottom
+                    runLog.debug(
+                        f"  - {potentialBlock} accounts for {heightOverlap/totalHeight * 100.0:<5.2f}% of the homogenized region"
                     )
+
                     if specialXSType is None:
                         sourceBlock = potentialBlock
                         specialXSType = sourceBlock.p.xsType
                     elif specialXSType == potentialBlock.p.xsType:
                         pass
                     else:
-                        runLog.error(
-                            "There are two special block XS types.  Not sure which to choose {} {}"
-                            "".format(sourceBlock, potentialBlock)
-                        )
-                        raise RuntimeError(
-                            "There are multiple special block XS types when there should only be one"
+                        msg = (
+                            f"Both {sourceBlock} and {potentialBlock} have conflicting XS types "
+                            f"and have either {[Flags.FUEL, Flags.CONTROL]} flags. One or the other flags "
+                            f"from these blocks should be removed to produce a uniform axial mesh for {newAssem}"
                         )
+                        runLog.error(msg)
+                        raise RuntimeError(msg)
+                elif potentialBlock == sourceBlock:
+                    totalHeight = topMeshPoint - bottom
+                    runLog.debug(
+                        f"  - {sourceBlock} accounts for {sourceBlockHeightOverlap/totalHeight * 100.0:<5.2f}% of the homogenized region"
+                    )
+                else:
+                    totalHeight = topMeshPoint - bottom
+                    runLog.debug(
+                        f"  - {potentialBlock} accounts for {heightOverlap/totalHeight * 100.0:<5.2f}% of the homogenized region"
+                    )
+
+            newXSType = sourceBlock.p.xsType if specialXSType is None else specialXSType
```

Comment:
At the very least, the title of this PR/commit will need to change. This is more than just debug statements

Reply:
My understanding is that this should behave the same as before but it's more explicit rather than relying on the assignment of the XS type just from the deepcopy. Maybe this logic is too complex as it is though and hard to understand.

Reply:
Okay. if it's "supposed" to come from the copied assem, I'm happy